src/common/market/order.js

```diff
@@ -1034,3 +1081,89 @@ export const createRequestorder = async (
     tag: await tagSchema().validate(tag),
   };
 };
+
+export const estimateMatchOrders = async (
+  contracts,
+  voucherHubAddress,
+  appOrder,
+  datasetOrder = NULL_DATASETORDER,
+  workerpoolOrder,
+  requestOrder,
+  useVoucher,
+) => {
+  const [vAppOrder, vDatasetOrder, vWorkerpoolOrder, vRequestOrder] =
+    await Promise.all([
+      signedApporderSchema().validate(appOrder),
+      signedDatasetorderSchema().validate(datasetOrder),
+      signedWorkerpoolorderSchema().validate(workerpoolOrder),
+      signedRequestorderSchema().validate(requestOrder),
+    ]);
+  const matchableVolume = await getMatchableVolume(
+    contracts,
+    vAppOrder,
+    vDatasetOrder,
+    vWorkerpoolOrder,
+    vRequestOrder,
+  );
+  const totalCost = getMatchOrdersTotalCost(
+    matchableVolume,
+    vAppOrder.appprice,
+    vDatasetOrder.datasetprice,
+    vWorkerpoolOrder.workerpoolprice,
+  );
+  let sponsoredCost = new BN(0);
+  if (useVoucher) {
+    const voucherAddress = await fetchVoucherAddress(
+      contracts,
+      voucherHubAddress,
+      requestOrder.requester,
+    );
+    if (!voucherAddress) {
+      return { total: totalCost, sponsored: sponsoredCost };
+    }
+    const voucherContract = await getVoucherContract(contracts, voucherAddress);
+
+    const [balance, voucherTypeId] = await Promise.all([
+      voucherContract.getBalance(),
+      voucherContract.getType(),
+    ]);
```

Comment:
Vouchers have an expiration date, we must check if the expiration date is not reached.
If the expiration date is reached, the sponsored amount is 0.

\+ add a test for this case